Make it work for uppercase suffixes of image files

hnishi opened the pull request:
I tried to train a model using my custom dataset which was annotated with [cvat](https://github.com/openvinotoolkit/cvat) .

I got an error due to uppercases "PNG" of my image files.
 (The image files had been generated by cvat.)

```
Could not read label 'data/custom/labels/frame_003400.PNG'.
Could not read label 'data/custom/labels/frame_000600.PNG'.
Could not read label 'data/custom/labels/frame_003700.PNG'.
Training Epoch 0:   0%|                                                                                             | 0/6 [00:02<?, ?it/s]
Traceback (most recent call last):
  File "train.py", line 105, in <module>
    for batch_i, (_, imgs, targets) in enumerate(tqdm.tqdm(dataloader, desc=f"Training Epoch {epoch}")):
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/tqdm/std.py", line 1178, in __iter__
Could not read label 'data/custom/labels/frame_003300.PNG'.
Could not read label 'data/custom/labels/frame_000800.PNG'.
Could not read label 'data/custom/labels/frame_000700.PNG'.
    for obj in iterable:
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 517, in __next__
Could not read label 'data/custom/labels/frame_003200.PNG'.
    data = self._next_data()
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1199, in _next_data
    return self._process_data(data)
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 1225, in _process_data
    data.reraise()
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/torch/_utils.py", line 429, in reraise
    raise self.exc_type(msg)
ValueError: Caught ValueError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/torch/utils/data/_utils/worker.py", line 202, in _worker_loop
    data = fetcher.fetch(index)
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/.venv/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 47, in fetch
    return self.collate_fn(data)
  File "/Users/Hiroshi.Nishigami/work/PyTorch-YOLOv3/utils/datasets.py", line 129, in collate_fn
    paths, imgs, bb_targets = list(zip(*batch))
ValueError: not enough values to unpack (expected 3, got 0)
```

Flova commented:
First of all thank you for your pull request!
Your solution looks good to me, but I raised the idea to rebuild this part in a cleaner way using the python path utilizes from the standard library. We could use the `basename` feature to strip the last file extension to replace it with the `.txt`. This should be more reliable to names like `PNGfoobar.cvat.png`. We could also use these utilities to find the parent folder of the images to replace it with the labels folder. At the moment these replacement is done on the whole path and paths like `/home/doe/images_version_5_dataset/images/foobar.png` would be replaced to `/home/doe/labels_version_5_dataset/labels/foobar.png`. This shouldn't be the case.

hnishi commented:
Thank you for your comments!

> I raised the idea to rebuild this part in a cleaner way using the python path utilizes from the standard library. 

I agree.

I modified the code by using the os.path module (4676740).

How about this modification?

I also tested it as follows

```python
>>> import os
>>> path = "/home/doe/images_version_5_dataset/images/PNG_foobar.png"
>>> label_file = os.path.join(os.path.split(os.path.dirname(path))[0], 'labels')
>>> label_file = os.path.join(label_file, os.path.basename(path))
>>> label_file = os.path.splitext(label_file)[0] + '.txt'
>>> print(label_file)
/home/doe/images_version_5_dataset/labels/PNG_foobar.txt
```

Flova reviewed (approved):
Thanks for implementing this! I really like your implementation.

Flova commented:
This breaks some paths like the coco ones which are split like this `('/home/florian/Projekt/PyTorch-YOLOv3/data/coco128/images', 'train2017')`.
This seems to be very hard to solve using the path lib. I opted for this solution its not as pretty, but we need to support coco and so on:
```python
image_dir = os.path.dirname(path) 
label_dir = "labels".join(image_dir.rsplit("images", 1))
assert label_dir != image_dir, \
     f"Image path must contain a folder named 'images'! \n'{image_dir}'"
label_file = os.path.join(label_dir, os.path.basename(path))
label_file = os.path.splitext(label_file)[0] + '.txt'
```
